types/lodash/lodash-tests.ts

```diff
@@ -4198,6 +4198,12 @@ fp.now(); // $ExpectType number
     if (_.isEmpty(foo)) {
         foo.bar = "baz";
     }
+
+    let unionFromEmptyAndNonEmpty: null | undefined | 0 | '' | object | boolean
+    unionFromEmptyAndNonEmpty = [null, undefined, 0 as const, '' as const, { test: 1 }, false, true][Math.round(Math.random() * 5)]
+    if (!_.isEmpty(unionFromEmptyAndNonEmpty)) {
```

Comment:
Test both branches (true and false). The object should also appear in the true branch.